zone-service/src/main/java/org/cloudfoundry/identity/uaa/zone/OrchestratorZoneService.java

```diff
@@ -88,13 +171,305 @@ private ConnectionDetails buildConnectionDetails(String zoneName, IdentityZone i
+            throw new OrchestratorZoneServiceException("Unexpected exception while create identity provider for " +
+                                                       "subdomain : ." + created.getSubdomain() + " Exception is : " + e.getMessage());
+        }
+    }
+
+    private IdentityZone createIdentityZone(IdentityZone identityZone)
+        throws OrchestratorZoneServiceException {
+        IdentityZone created = null;
+        try {
+            created = zoneProvisioning.create(identityZone);
+        } catch (Exception e) {
+            logger.error("Unable to create identity zone for subdomain : " + identityZone.getSubdomain() + " Exception is : " + e.getMessage());
+            throw new OrchestratorZoneServiceException("Unexpected exception while create identity zone for " +
+                                                       "subdomain : ." + identityZone.getSubdomain() + " Exception is : " + e.getMessage());
+        }
+        return created;
+    }
+
+    private IdentityZone geIdentityZone(String subdomain, String name, String id, String zoneSigningKey) throws OrchestratorZoneServiceException {
+        IdentityZone identityZone = new IdentityZone();
+        identityZone.setId(id);
+        identityZone.setName(name);
+        identityZone.setSubdomain(subdomain);
+        setTokenPolicy(zoneSigningKey, identityZone);
+        setSamlConfig(subdomain, identityZone);
+        identityZone.getConfig().getLinks().getLogout().setWhitelist(createDeploymentSpecificLogoutWhiteList());
+        return identityZone;
+    }
+
+    private void setSamlConfig(String subdomain, IdentityZone identityZone) throws OrchestratorZoneServiceException {
+        try {
+            identityZone.getConfig().setSamlConfig(createSamlConfig(subdomain));
+        } catch (Exception e) {
+            logger.error("Unable to create saml config for subdomain: " + subdomain + " Exception is : " + e.getMessage());
+            throw new OrchestratorZoneServiceException("Unexpected exception while create saml config for " +
+                                                       "subdomain: ." + subdomain + " Exception is : " + e.getMessage());
+        }
+    }
+
+    private void setTokenPolicy(String zoneSigningKey, IdentityZone identityZone) {
+        String activeKeyId = new RandomValueStringGenerator(5).generate();
+        Map<String, String> keys = getKeys(zoneSigningKey, activeKeyId);
+        TokenPolicy tokenPolicy = new TokenPolicy();
+        tokenPolicy.setActiveKeyId(activeKeyId);
+        tokenPolicy.setKeys(keys);
+        identityZone.getConfig().setTokenPolicy(tokenPolicy);
+    }
+
+    private String createSigningKey(String subdomain) throws OrchestratorZoneServiceException {
+        String zoneSigningKey = "";
+        try {
+            zoneSigningKey = createSigningKey();
+        } catch (Exception e) {
+            logger.error("Unable to create signingKey for subdomain: " + subdomain + " Exception is : " + e.getMessage());
```

Comment:
Log entire stack trace instead of printing only message. This will helpful for debug.

Reply:
Done. Thanks!